### ISSUE TITLE
DDC-3671: Duplicated unique indexes (@UniqueConstraint annotation)

### ISSUE THREAD

**Issue by doctrinebot** [doctrine/doctrine2#4502]
Jira issue originally created by user webimpress:

In this commit:
https://github.com/doctrine/dbal/commit/fc8beca08600b44e46f7b82998adbcc6d52c31f1
was changed to not checking for if index already exists. After that occur problem with duplicated unique keys (for example migrations try add unique index on the same columns with different index name).

This is an example model:

```
/****
 * @Entity
 * @Table(name="double*uniq_index*table", uniqueConstraints={
 *   @UniqueConstraint(name="uniq_hash", columns={"hash"})
 * })
 */
class DoubleUniqIndexModel
{
    /*** @Id @Column **/
    private $id;

    /*** @Column(name="hash", type="string", length=8, nullable=false, unique=true) **/
    private $hash;
}
```

and the basic test:

```
    public function testDoubleUniqIndex()
    {
        $em = $this->_getTestEntityManager();
        $schemaTool = new SchemaTool($em);

        $classes = array(
            $em->getClassMetadata(*_NAMESPACE_* . '\\DoubleUniqIndexModel'),
        );

        $schema = $schemaTool->getSchemaFromMetadata($classes);

        $this->assertTrue($schema->hasTable('double*uniq_index*table'));
        $this->assertEquals(2, count($schema->getTable('double*uniq_index*table')->getIndexes()));
    }
```

(I've added it into: `tests/Doctrine/Tests/ORM/Tools/SchemaToolTest.php`).

Test failures, because generated schema contains three indexes (primary, and two unique on the same column with different names - one auto generated `uniq**` and one provided in annotations - `uniq*hash`).

**Comment by zerocrates** [comment]
This looks like it was closed by the merge of #4508

**Comment by Ocramius** [comment]
@zerocrates thanks, closing!
